File: Java_Advanced/gsapi/src/main/java/br/com/fiap/gs/gsapi/dto/alert/AlertableEventDTO.java
// Pacote: br.com.fiap.gs.gsapi.dto.alert
package br.com.fiap.gs.gsapi.dto.alert;

import java.time.OffsetDateTime;

public class AlertableEventDTO {
    private String eventId; // Ex: ID da EONET
    private String title;
    private String eventDate; // Pode ser String formatada ou OffsetDateTime se preferir converter
    private String link;      // Link para mais detalhes sobre o evento
    private String description; // Uma breve descrição do que será enviado no alerta

    // Construtores, Getters e Setters

    public AlertableEventDTO() {
    }

    public AlertableEventDTO(String eventId, String title, String eventDate, String link, String description) {
        this.eventId = eventId;
        this.title = title;
        this.eventDate = eventDate;
        this.link = link;
        this.description = description;
    }

    public String getEventId() {
        return eventId;
    }

    public void setEventId(String eventId) {
        this.eventId = eventId;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getEventDate() {
        return eventDate;
    }

    public void setEventDate(String eventDate) {
        this.eventDate = eventDate;
    }

    public String getLink() {
        return link;
    }

    public void setLink(String link) {
        this.link = link;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }
}

File: Advanced_Business_Development_with.NET/gsApi/gsApi/dto/request/AlertableEventDto.cs
// File: gsApi/dto/request/AlertableEventDto.cs
namespace gsApi.dto.request
{
    /// <summary>
    /// DTO contendo detalhes de um evento para fins de alerta.
    /// </summary>
    public class AlertableEventDto
    {
        /// <summary>
        /// ID do evento (ex: ID da EONET).
        /// </summary>
        public string? EventId { get; set; }

        /// <summary>
        /// Título do evento. Se for obrigatório para o alerta, marque como 'required'.
        /// </summary>
        public required string Title { get; set; } // Exemplo: Tornando o título obrigatório

        /// <summary>
        /// Data do evento (formatada como string).
        /// </summary>
        public string? EventDate { get; set; }

        /// <summary>
        /// Link para mais informações sobre o evento.
        /// </summary>
        public string? Link { get; set; }

        /// <summary>
        /// Descrição breve do evento.
        /// </summary>
        public string? Description { get; set; }
    }
}